src/gui/qgsmapcanvas.cpp

```diff
@@ -147,6 +147,8 @@ QgsMapCanvas::QgsMapCanvas( QWidget *parent )
   mSettings.setFlag( QgsMapSettings::DrawEditingInfo );
   mSettings.setFlag( QgsMapSettings::UseRenderingOptimization );
   mSettings.setFlag( QgsMapSettings::RenderPartialOutput );
+  mSettings.setDestinationCrs( QgsProject::instance()->crs() );
+  mSettings.setEllipsoid( QgsProject::instance()->ellipsoid() );
```

Comment:
No - it's QgsProject::ellipsoidChanged. The project CRS -> canvas CRS handling is mature and should not be changed here.